controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java

```diff
@@ -78,7 +78,9 @@ public void testTriggerFailing() {
         tester.deployAndNotify(app, applicationPackage, true, JobType.systemTest);
 
         // staging-test times out and is retried
-        tester.buildService().takeJobsToRun();
+        synchronized (tester.buildService()) {
```

Comment:
Do we need all this synchronization? We don't run individual tests in parallel as far as I know.

Reply:
Urgh, this was not intentional; the MockBuildSystem itself uses a synchronised list now.
The DeploymentTrigger does things to its BuildSystem in parallel, and so any implementations need to be thread-safe. I'll add this in the interface doc.